File: TransCoder_tokenized_test_set_functions/python/HEIGHT_N_ARY_TREE_PARENT_ARRAY_GIVEN.py
def findHeight ( parent, n ) :
    res = 0
    for i in range ( n ) :
        p = i
        current = 1
        while ( parent [ p ] != - 1 ) :
            current += 1
            p = parent [ p ]
        res = max ( res, current )
    return res

File: TransCoder_tokenized_test_set_functions/java/HEIGHT_N_ARY_TREE_PARENT_ARRAY_GIVEN.java
static int findHeight ( int [ ] parent, int n ) {
  int res = 0;
  for ( int i = 0;
  i < n;
  i ++ ) {
    int p = i, current = 1;
    while ( parent [ p ] != - 1 ) {
      current ++;
      p = parent [ p ];
    }
    res = Math . max ( res, current );
  }
  return res;
}